pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ActivityInstanceStatusDao.java

```diff
@@ -375,6 +372,37 @@ default void runPostActivityStatusChangeHooks(
         }
     }
 
+    private void processAnnouncementAction(Handle handle, long participantId, long studyId, EventConfigurationDto eventConfig) {
+        try {
+            UserAnnouncementDao announcementDao = getUserAnnouncementDao();
+            if (eventConfig.getAnnouncementCreateForProxies() == true) {
+                List<Governance> governances = handle.attach(UserGovernanceDao.class)
+                        .findActiveGovernancesByParticipantAndStudyIds(participantId, studyId)
+                        .collect(Collectors.toList());
+                if (!governances.isEmpty()) {
+                    for (var governance : governances) {
+                        long id = announcementDao.insert(governance.getProxyUserId(), studyId,
+                                eventConfig.getAnnouncementMsgTemplateId(), eventConfig.getAnnouncementIsPermanent());
+                        LOG.info("Created new announcement with id {} for proxy id {} (participant id {}) and study id {}",
+                                id, governance.getProxyUserId(), participantId, studyId);
+                    }
+                } else {
+                    LOG.error("Participant with id {} has no active proxies in study id {}."
+                            + " Announcement with event configuration id {} will not be created.",
+                            participantId, studyId, eventConfig.getEventConfigurationId());
+                }
+            } else {
+                long id = announcementDao.insert(participantId, studyId,
+                        eventConfig.getAnnouncementMsgTemplateId(), eventConfig.getAnnouncementIsPermanent());
+                LOG.info("Created new announcement with id {} for participant id {} and study id {}", id, participantId, studyId);
+            }
+        } catch (Exception e) {
+            throw new DaoException(String.format(
```

Comment:
Is the goal here to add more context to the exception?  If so, I'm all for it.  Looking up the stack, it does appear that there would be very little context in the log, so this will be helpful.

Reply:
Yup. Given what's wrapped in the try/catch, most likely we'll get a SQL exception or similar which doesn't have context for user/study.